tasks/python_versions.yml

```diff
@@ -91,9 +88,8 @@
 
 - name: Set pyenv global version to '{{ pyenv_global }}'
   shell: >-
-    . {{ pyenv_root }}/.pyenvrc &&
-    pyenv global {{ pyenv_global }} &&
-    pyenv rehash
+    {{ pyenv_root }}/bin/pyenv global &&
```

Comment:
Looks like you've dropped `{{ pyenv_global }}` in here so the global version is not set correctly.